plugins/devtools-backend/src/service/router.ts

```diff
@@ -35,30 +35,66 @@ import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-
 import {
   DiscoveryService,
   HttpAuthService,
+  LifecycleService,
+  LoggerService,
   PermissionsService,
 } from '@backstage/backend-plugin-api';
+import { SignalService } from '@backstage/plugin-signals-node';
 
 /** @public */
 export interface RouterOptions {
   devToolsBackendApi?: DevToolsBackendApi;
-  logger: Logger;
+  logger: LoggerService;
   config: Config;
   permissions: PermissionsService;
   discovery: DiscoveryService;
   httpAuth?: HttpAuthService;
+  lifecycle?: LifecycleService;
+  signalService?: SignalService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, config, permissions } = options;
+  const { logger, config, permissions, lifecycle, signalService } = options;
 
   const { httpAuth } = createLegacyAuthAdapters(options);
 
   const devToolsBackendApi =
     options.devToolsBackendApi || new DevToolsBackendApi(logger, config);
 
+  if (lifecycle && signalService) {
+    let interval: NodeJS.Timeout | undefined;
+    lifecycle.addStartupHook(async () => {
+      interval = setInterval(async () => {
+        const info = await devToolsBackendApi.listInfo();
+        await signalService.publish<DevToolsInfo>({
+          channel: 'devtools:info',
+          message: info,
+          recipients: null,
+          // TODO: Add permissions from https://github.com/backstage/backstage/pull/23171
```

Comment:
It is a bit blocking. We do not want to send the signal without checking the access rights who can get those.

Reply:
Hmm, it's potentially problematic to need to apply permissions for a lot of signals, I could see that leading to scaling issues. Perhaps a fairly reasonable compromise could be to avoid signals with sensitive information altogether? Then again that might be very limiting to design around : /

Another option could be that we only apply permissions during subscription, rather that when receiving a signal?